drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -133,6 +132,28 @@ class DragAndDropBlock(
         enforce_type=True,
     )
 
+    showanswer = String(
+        display_name=_("Show Answer"),
+        help=_("Defines when to show the answer to the problem. "
+               "A default value can be set in Advanced Settings."),
+        scope=Scope.settings,
+        default=SHOWANSWER.FINISHED,
+        values=[
+            {"display_name": _("Always"), "value": SHOWANSWER.ALWAYS},
+            {"display_name": _("Answered"), "value": SHOWANSWER.ANSWERED},
+            {"display_name": _("Attempted or Past Due"), "value": SHOWANSWER.ATTEMPTED},
+            {"display_name": _("Closed"), "value": SHOWANSWER.CLOSED},
+            {"display_name": _("Finished"), "value": SHOWANSWER.FINISHED},
+            {"display_name": _("Correct or Past Due"), "value": SHOWANSWER.CORRECT_OR_PAST_DUE},
+            {"display_name": _("Past Due"), "value": SHOWANSWER.PAST_DUE},
+            {"display_name": _("Never"), "value": SHOWANSWER.NEVER},
+            {"display_name": _("After All Attempts"), "value": SHOWANSWER.AFTER_ALL_ATTEMPTS},
+            {"display_name": _("After All Attempts or Correct"), "value": SHOWANSWER.AFTER_ALL_ATTEMPTS_OR_CORRECT},
+            {"display_name": _("Attempted"), "value": SHOWANSWER.ATTEMPTED_NO_PAST_DUE},
+        ],
+        enforce_type=True,
+    )
```

Comment:
We don't need to introduce a new settings for this XBlock. The value set in the "Advanced Settings" of the course is already available in the XBlock's `showanswer` attribute and can be accessed using `self.xblock`. So this can be dropped.

Reply:
If I drop the above setting for  ```showanswer```,  tests breaks and I got an error

```python
  File "/home/abdullah/workspace/opencraft/src/xblock-drag-and-drop-v2/drag_and_drop_v2/drag_and_drop_v2.py", line 413, in student_view_data
    "show_answer_status": self.showanswer,
AttributeError: 'DragAndDropBlockWithMixins' object has no attribute 'showanswer'
```
Is there another way to access ```showanswer`` XBlock setting?

Thanks

Reply:
@abodacs Interestingly the attribute seems to load the right values from the advanced settings during runtime. I tried to see if I can find when/where this attribute is being added to the XBlock, but couldn't. I see a couple ways to get around this:
1. Mock `showanswer` in the tests
2. Define `showanswer` as a property in the XBlock and make it return the value by fetching the course settings from the runtime. But I think we might be unnecessarily re-implementing an existing functionality for the sake of tests.
3. Define a wrapper around the auto-injected `showanswer` like below and using that
```
@property
def show_answer(self):
    return getattr(self, 'showanswer', 'answered')
```

But, I am not sure which is the right way to go.

@Agrendalath Can you help here?

Reply:
@tecoholic, @abodacs,

This attribute comes from the [InheritanceMixin](https://github.com/openedx/edx-platform/blob/fcb594d6352b93e53aa89fcc758e18ae709752fc/xmodule/modulestore/inheritance.py#L86-L98). This mixin, [along with some other mixins](https://github.com/openedx/edx-platform/blob/fcb594d6352b93e53aa89fcc758e18ae709752fc/cms/envs/common.py#L932-L941), is usually injected [while constructing an XBlock](https://github.com/openedx/edx-platform/blob/fcb594d6352b93e53aa89fcc758e18ae709752fc/xmodule/modulestore/split_mongo/caching_descriptor_system.py#L232-L242).

That said, I believe it's a good idea to introduce the `showanswer` field directly in the XBlock - even though we define default behavior for the course, we may still want to override it for particular XBlocks.

Reply:
@Agrendalath I see. Thank you for the references. I have needed them multiple times before as well, and have simply settled to using a debugger to inspect the final xblock. I am going to bookmark these for future uses.

@abodacs I am sorry for misleading you on this. You have been on the right path :+1: